File: js/n-ary_tree _level_order_traversal.js
/**
 * // Definition for a Node.
 * function Node(val,children) {
 *    this.val = val;
 *    this.children = children;
 * };
 */

/**
 * @param {Node|null} root
 * @return {number[][]}
 */
 var levelOrder = function(root) {
    if (root == null) {
        return [];
    }
    
    let queue = [];
    queue.push({level: 0, node: root});
    
    let result = [];
    
    while(queue.length > 0) {
        let curr = queue.shift();
        let level = curr.level;
        
        if (result[level] == null) {
            result[level] = [];
        }
        result[level].push(curr.node.val);
        
        for(let child of curr.node.children) {
            queue.push({level: level + 1, node: child});
        }
    }
    return result;
};

File: java/n-ary_tree _level_order_traversal.java
/*
// Definition for a Node.
class Node {
    public int val;
    public List<Node> children;

    public Node() {}

    public Node(int _val) {
        val = _val;
    }

    public Node(int _val, List<Node> _children) {
        val = _val;
        children = _children;
    }
};
*/

class Solution {
    public List<List<Integer>> levelOrder(Node root) {
        Queue <Node> queue = new LinkedList<>();
        
        List <List<Integer>> ans = new ArrayList<>();
        
        if (root == null) {
            return ans;
        }
        queue.add(root);
        
        while (!queue.isEmpty()) {
            int sz = queue.size();
            
            List <Integer> list = new ArrayList<>();
            for (int i = 0; i < sz; i++) {
                Node curr = queue.poll();
                
                list.add(curr.val);
                for (Node c : curr.children) {
                    queue.add(c);
                }
            }
            
            ans.add(list);
        }
        
        return ans;
    }
}